devito/types/basic.py

```diff
@@ -549,7 +549,7 @@ def __new__(cls, *args, **kwargs):
     __hash__ = Uncached.__hash__
 
 
-class Scalar(Symbol, ArgProvider):
+class Scalar(Symbol):
```

Comment:
Well that was intentional 
- Scalar carry data
- Symbol don't carry data

So the second one is not an arg provider as it does not have values

Reply:
It's to remove the extraneous no-op methods in `Thickness`. Maybe `DataSymbol` should get `ArgProvider` instead then?